.github/workflows/ci-test.yml

```diff
@@ -91,90 +95,164 @@ jobs:
           shellcheck docker/build_push.sh && \
           shellcheck docker/dockerhub_to_novi_ecr.sh
       - name: docker lints
-        run: |
-          hadolint docker/ci/github/Dockerfile && \
-          hadolint docker/ci/arch/Dockerfile && \
-          hadolint docker/ci/alpine/Dockerfile && \
-          hadolint docker/ci/centos/Dockerfile
+        uses: ./.github/actions/docker-checks
 
   lint:
     runs-on: ubuntu-latest-xl
     timeout-minutes: 30
-    continue-on-error: false
     needs: prepare
     if: ${{ needs.prepare.outputs.test-rust == 'true' }}
     container:
       image: libra/build_environment:github-1
+      volumes:
+        - "/home/runner/work/diem/diem:/opt/git/diem"
     steps:
       - uses: actions/checkout@v2
         with:
           ref: ${{ github.event.pull_request.head.sha }}
       - uses: ./.github/actions/build-setup
+      - uses: actions/cache@v2
+        with:
+          path: "/opt/cargo/git\n/opt/cargo/registry\n/opt/cargo/.package-cache"
+          key: crates-1-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
```

Comment:
n00b question - Is the 1 in `crates-1-*` the manual cache index so that we can invalidate them if need to?

Reply:
Good eyes, this is residual testing that I should clean up.

Reply:
At least I was consistent.